File: Foundation/Scratch/String/ques2.cpp
// ques:
// input: abaabbbccddeffghhh
// output: a3b4c2d2e1f2g1h3

#include<iostream>
#include<vector>
using namespace std;
string stringCompress(string str){
        vector<int> freq(26,0);
        for(int i = 0; i < str.length(); i++){
            freq[str[i] - 'a']++;
        }
        string st = "";
        for(int i = 0; i < freq.size(); i++){
            if(freq[i] > 0){
                st += (i+'a');
                st += to_string(freq[i]);
            }
        }
        return st;
    }
int main(){
    string str = "abaabbbccddeffghhh";
    cout<<stringCompress(str);
    return 0;
}

File: Foundation/Scratch/String/ques2.java
// ques:
// input: abaabbbccddeffghhh
// output: a3b4c2d2e1f2g1h3

public class ques2{
    public static String stringCompress(String str){
        int[] freq = new int[26];
        for(int i = 0; i < str.length(); i++){
            freq[str.charAt(i) - 'a']++;
        }
        StringBuilder sb = new StringBuilder();
        for(int i = 0; i < freq.length; i++){
            if(freq[i] > 0){
                sb.append((char)(i+'a'));
                sb.append(freq[i]);
            }
        }
        return sb.toString();
    }
    public static void main(String[] args){
        String str = "abaabbbccddeffghhh";
        System.out.println(stringCompress(str));
    }
}
